packages/2d/src/components/SVG.ts

```diff
@@ -0,0 +1,387 @@
+      const center = new BBox(0, 0, width, height).center;
+      const rect = new Rect({
+        width,
+        height,
+        ...style,
+      });
+      this.applyTransformToShape(
+        rect,
+        transformMatrix.translate(center.x, center.y),
+      );
+      yield rect;
+    }
+  }
+
+  private isNodeEqual(aShape: Node, bShape: Node): boolean {
+    if (aShape.constructor !== bShape.constructor) return false;
+    if (
+      aShape instanceof Path &&
+      bShape instanceof Path &&
+      aShape.data() !== bShape.data()
+    )
+      return false;
+
+    return true;
+  }
+
+  private diffSVG(from: ParsedSVG, to: ParsedSVG): SVGDiff {
+    const diff: SVGDiff = {
+      fromSize: from.size,
+      toSize: to.size,
+      inserted: [],
+      deleted: [],
+      transformed: [],
+    };
+
+    const aNodes = from.nodes;
+    const bNodes = to.nodes;
+    const aLength = aNodes.length;
+    const bLength = bNodes.length;
+    let aIndex = 0;
+    let bIndex = 0;
+
+    diffSequence(
+      aLength,
+      bLength,
+      (aIndex, bIndex) => {
+        return this.isNodeEqual(aNodes[aIndex], bNodes[bIndex]);
+      },
+      (nCommon, aCommon, bCommon) => {
+        if (aIndex !== aCommon)
+          diff.deleted.push(...aNodes.slice(aIndex, aCommon));
+        if (bIndex !== bCommon)
+          diff.inserted.push(...bNodes.slice(bIndex, bCommon));
+
+        aIndex = aCommon;
+        bIndex = bCommon;
+        for (let x = 0; x < nCommon; x++) {
+          diff.transformed.push({
+            from: aNodes[aIndex],
+            to: bNodes[bIndex],
+          });
+          aIndex++;
+          bIndex++;
+        }
+      },
+    );
+
+    if (aIndex !== aLength) diff.deleted.push(...aNodes.slice(aIndex));
+
+    if (bIndex !== bNodes.length) diff.inserted.push(...bNodes.slice(bIndex));
+
+    diff.deleted = diff.deleted.filter(aNode => {
+      const bIndex = diff.inserted.findIndex(bNode =>
+        this.isNodeEqual(aNode, bNode),
+      );
+      if (bIndex >= 0) {
+        const bNode = diff.inserted[bIndex];
+        diff.inserted.splice(bIndex, 1);
+        diff.transformed.push({
+          from: aNode,
+          to: bNode,
+        });
+
+        return false;
+      }
+
+      return true;
+    });
+    return diff;
+  }
+
+  private cloneNodeExact(node: Node) {
+    const props: ShapeProps = {
+      position: node.position(),
+      scale: node.scale(),
+      rotation: node.rotation(),
+    };
+    if (node instanceof Layout) {
+      props.size = node.size();
+    }
+    return node.clone(props);
+  }
+
+  public *tweenSVG(svg: string, time: number) {
+    const newSVG = this.parseSVG(svg);
+    const diff = this.diffSVG(this.parsed(), newSVG);
+    const transformed = diff.transformed.map(({from, to}) => ({
+      from: this.cloneNodeExact(from),
+      current: from,
+      to,
+    }));
+
+    for (const node of diff.inserted) this.wrapper.add(node);
```

Comment:
Will this insert them in the correct order?
I know that we reset the wrapper children at the end of the tween but if there are overlapping elements in the svg couldn't that cause visible glitches during tweening?